src/app/route/CartPage/CartPage.container.js

```diff
@@ -43,6 +83,34 @@ export class CartPageContainer extends PureComponent {
         this._changeHeaderState();
     }
 
+    componentDidUpdate(prevProps) {
+        const {
+            changeHeaderState,
+            totals: { items_qty },
+            headerState,
+            headerState: { name }
+        } = this.props;
+
+        const {
+            totals: { items_qty: prevItemsQty },
+            headerState: { name: prevName }
+        } = prevProps;
+
+        if (name !== prevName) {
+            if (name === CART) {
+                this._changeHeaderState();
+            }
+        }
+
+        if (items_qty !== prevItemsQty) {
+            const title = `${ items_qty || '0' } Items`;
+            changeHeaderState({
+                ...headerState,
+                title
```

Comment:
This might not work, as it pushes state into history, prefer "force" pushing, to override. I think I will take this code and migrate it into redesign manually.